src/daft-image/src/series.rs

```diff
@@ -0,0 +1,159 @@
+    s: &Series,
+    raise_error_on_failure: bool,
+    mode: Option<ImageMode>,
+) -> DaftResult<Series> {
+    match s.data_type() {
+        DataType::Binary => image_decode_impl(s.binary()?, raise_error_on_failure, mode)
+            .map(|arr| arr.into_series()),
+        dtype => Err(DaftError::ValueError(format!(
+            "Decoding in-memory data into images is only supported for binary arrays, but got {}",
+            dtype
+        ))),
+    }
+}
+
+pub fn encode(s: &Series, image_format: ImageFormat) -> DaftResult<Series> {
+    match s.data_type() {
+        DataType::Image(..) => Ok(s
+            .downcast::<ImageArray>()?
+            .encode(image_format)?
+            .into_series()),
+        DataType::FixedShapeImage(..) => Ok(s
+            .downcast::<FixedShapeImageArray>()?
+            .encode(image_format)?
+            .into_series()),
+        dtype => Err(DaftError::ValueError(format!(
+            "Encoding images into bytes is only supported for image arrays, but got {}",
+            dtype
+        ))),
+    }
+}
+
+pub fn resize(s: &Series, w: u32, h: u32) -> DaftResult<Series> {
+    match s.data_type() {
+        DataType::Image(mode) => {
+            let array = s.downcast::<ImageArray>()?;
+            match mode {
+                // If the image mode is specified at the type-level (and is therefore guaranteed to be consistent
+                // across all images across all partitions), store the resized image in a fixed shape image array,
+                // since we'll have homogeneous modes, heights, and widths after resizing.
+                Some(mode) => Ok(array
+                    .resize_to_fixed_shape_image_array(w, h, mode)?
+                    .into_series()),
+                None => Ok(array.resize(w, h)?.into_series()),
+            }
+        }
+        DataType::FixedShapeImage(..) => Ok(s
+            .downcast::<FixedShapeImageArray>()?
+            .resize(w, h)?
+            .into_series()),
+        _ => Err(DaftError::ValueError(format!(
+            "datatype: {} does not support Image Resize. Occurred while resizing Series: {}",
+            s.data_type(),
+            s.name()
+        ))),
+    }
+}
+
+pub fn crop(s: &Series, bbox: &Series) -> DaftResult<Series> {
+    let bbox_type = DataType::FixedSizeList(Box::new(DataType::UInt32), 4);
+    let bbox = bbox.cast(&bbox_type)?;
+    let bbox = bbox.fixed_size_list()?;
+
+    match &s.data_type() {
+        DataType::Image(_) => s
+            .downcast::<ImageArray>()?
+            .crop(bbox)
+            .map(|arr| arr.into_series()),
+        DataType::FixedShapeImage(..) => s
+            .fixed_size_image()?
+            .crop(bbox)
+            .map(|arr| arr.into_series()),
+        dt => Err(DaftError::ValueError(format!(
+            "Expected input to crop to be an Image type, but received: {}",
+            dt
+        ))),
+    }
+}
+
+pub fn to_mode(s: &Series, mode: ImageMode) -> DaftResult<Series> {
```

Comment:
Any chance we could generate some documentation for these functions given that they're the public API for this crate? 😛 

Copilot makes that super easy haha